code/_globalvars/traits.dm

```diff
@@ -180,3 +187,19 @@ GLOBAL_LIST(trait_name_map)
 		for(var/tname in GLOB.traits_by_type[key])
 			var/val = GLOB.traits_by_type[key][tname]
 			.[val] = tname
+
+GLOBAL_LIST_INIT(movement_type_trait_to_flag, list(
+	TRAIT_MOVE_GROUND = GROUND,
+	TRAIT_MOVE_FLYING = FLYING,
+	TRAIT_MOVE_VENTCRAWLING = VENTCRAWLING,
+	TRAIT_MOVE_FLOATING = FLOATING,
+	TRAIT_MOVE_PHASING = PHASING
+	))
+
+GLOBAL_LIST_INIT(movement_type_trait_add_signals, set_movement_type_trait_signals(SIGNAL_ADDTRAIT("")))
+GLOBAL_LIST_INIT(movement_type_trait_remove_signals, set_movement_type_trait_signals(SIGNAL_REMOVETRAIT("")))
+
+/proc/set_movement_type_trait_signals(signal_prefix)
+	. = list()
+	for(var/trait in GLOB.movement_type_trait_to_flag)
+		. += "[signal_prefix][trait]"
```

Comment:
This seems like a flimsy exploitation of how SIGNAL_ADDTRAIT/SIGNAL_REMOVETRAIT work internally--there's no guarantee that it'll be in this format forever. 

I think you should change the movement_type_trait_add_signals/movement_type_trait_remove_signals to be lazy lists, and have a wrapper to get them (like `get_movement_type_trait_add_signals`) that will dynamically set those lists and use `SIGNAL_ADDTRAIT(...)` and `SIGNAL_REMOVETRAIT(...)` how they were intended to be used.